<a href="https://colab.research.google.com/github/anabiarochar/Mestrado_AlgoritmosProgramacao/blob/main/Projeto1/Sele%C3%A7%C3%A3o.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [22]:
import requests
import time

# 1. Obter os dados do GitHub
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/refs/heads/main/Projeto1/Vetores/vetor_1000_decrescente.txt?token=GHSAT0AAAAAADXG3NI5M2O4MPNENAJ74TN62NNF5OQ"

try:
    response = requests.get(url)
    content = response.text

    # Lendo o arquivo separado por vírgulas
    vetor = [int(x.strip()) for x in content.split(',') if x.strip()]

    print(f"Sucesso! Vetor carregado com {len(vetor)} elementos.")

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")
    vetor = []

# 2. Implementação do Método de Seleção (Selection Sort)
def selection_sort(lista_original):
    # Criamos uma cópia para preservar o vetor original
    arr = lista_original.copy()
    n = len(arr)

    for i in range(n):
        # Encontrar o índice do menor elemento no restante do vetor
        indice_min = i
        for j in range(i + 1, n):
            if arr[j] < arr[indice_min]:
                indice_min = j

        # Trocar o menor elemento encontrado com o elemento da posição i
        arr[i], arr[indice_min] = arr[indice_min], arr[i]

    return arr

# 3. Execução e Medição de Tempo
if vetor:
    print("\nIniciando ordenação por Selection Sort...")

    # Início do cronômetro de alta precisão
    inicio = time.perf_counter()

    vetor_ordenado = selection_sort(vetor)

    # Fim do cronômetro
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print("--- Resultado ---")
    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(f"Primeiros 10 elementos ordenados: {vetor_ordenado[:10]}")
    print(f"Últimos 10 elementos ordenados: {vetor_ordenado[-10:]}")
else:
    print("Não foi possível processar o vetor.")

Sucesso! Vetor carregado com 1000 elementos.

Iniciando ordenação por Selection Sort...
--- Resultado ---
Tempo de execução: 0.047908 segundos
Primeiros 10 elementos ordenados: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Últimos 10 elementos ordenados: [991, 992, 993, 994, 995, 996, 997, 998, 999, 1000]



###Aleatório 1000:
Tentativa 1 - Tempo de execução: 0.025019 segundos   
Tentativa 2 - Tempo de execução: 0.047944 segundos  
Tentativa 3 - Tempo de execução: 0.051149 segundos   
Tentativa 4 - Tempo de execução: 0.025113 segundos

###Aleatório 10000:
Tentativa 1 - Tempo de execução: 2.562468 segundos  
Tentativa 2 - Tempo de execução: 2.686351 segundos  
Tentativa 3 - Tempo de execução: 2.500412 segundos  
Tentativa 4 - Tempo de execução: 3.145576 segundos

###Aleatório 100000:
Tentativa 1 - Tempo de execução: 316.358742 segundos   
Tentativa 2 - Tempo de execução:  
Tentativa 3 - Tempo de execução:  
Tentativa 4 - Tempo de execução:  

###Crescente 1000:
Tentativa 1 - Tempo de execução: 0.035588 segundos  
Tentativa 2 - Tempo de execução: 0.024099 segundos  
Tentativa 3 - Tempo de execução: 0.025444 segundos  
Tentativa 4 - Tempo de execução: 0.024301 segundos  

###Crescente 10000:
Tentativa 1 - Tempo de execução: 2.503939 segundos  
Tentativa 2 - Tempo de execução: 2.555702 segundos  
Tentativa 3 - Tempo de execução: 3.585622 segundos  
Tentativa 4 - Tempo de execução: 3.645931 segundos  

###Crescente 100000:
Tentativa 1 - Tempo de execução: 286.392151 segundos  
Tentativa 2 - Tempo de execução:  
Tentativa 3 - Tempo de execução:  
Tentativa 4 - Tempo de execução:  

###Decrescente 1000:
Tentativa 1 - Tempo de execução: 0.062301 segundos  
Tentativa 2 - Tempo de execução: 0.025683 segundos  
Tentativa 3 - Tempo de execução: 0.026021 segundos  
Tentativa 4 - Tempo de execução: 0.047908 segundos  

###Decrescente 10000:
Tentativa 1 - Tempo de execução:  
Tentativa 2 - Tempo de execução:  
Tentativa 3 - Tempo de execução:  
Tentativa 4 - Tempo de execução:  

###Decrescente 100000:
Tentativa 1 - Tempo de execução:  
Tentativa 2 - Tempo de execução:  
Tentativa 3 - Tempo de execução:  
Tentativa 4 - Tempo de execução:  